### PR TITLE
Python 3k: another round of compatibility improvements

### PR DESCRIPTION
**Pull request by clebergnu**
The title says it all.  While this still doesn't bring full Python 3 compatibility (there's still quite a bit to be done) it improves things a little, while not breaking Python 2 code.

### CODE REVIEW

File: avocado/core/job.py [hunk 1]
```diff
@@ -319,10 +319,14 @@ def _log_avocado_version():
...
-            status, top_commit = commands.getstatusoutput(cmd)
+            cmd = "git show --summary --pretty='%H'"
+            result = process.run(cmd, ignore_status=True)
+            status = result.exit_status
+            top_commit = result.stdout.splitlines()[0]
             cmd2 = "git rev-parse --abbrev-ref HEAD"
-            status2, branch = commands.getstatusoutput(cmd2)
+            result2 = process.run(cmd2)
+            status2 = result2.exit_status
+            branch = result2.stdout
```
Comment (lmr) [inline avocado/core/job.py]: OK

File: avocado/core/output.py [hunk 2]
```diff
@@ -441,7 +441,7 @@ def reconfigure(args):
         try:
             add_log_handler(name, logging.StreamHandler, STD_OUTPUT.stdout,
                             level)
-        except ValueError, details:
+        except ValueError as details:
```
Comment (apahim) [inline avocado/core/output.py]: I see other occurrences in `plugins/tap.py` and `utils/lv_utils.py`.

**Comment by lmr** [comment]
I thought the exception changes were (semi)automatically generated.

**Comment by clebergnu** [comment]
@lmr not really.  I was trying to make that the case, but hit some roadblocks.  @apahim will send a v2.  Thank you guys for the review.

**Comment by clebergnu** [comment]
@apahim actually there are a few more:

```
selftests/functional/test_lv_utils.py:            except BaseException, details:
selftests/functional/test_output.py:        except Exception, details:
```

Fixing all of them.

**Comment by clebergnu** [comment]
Closing in favor of #1860

**Pull request closed without merging**